dataset_api.md

```diff
@@ -0,0 +1,155 @@
+Incorporate the changes proposed from Martynas, with the exception of graphs(), which would now return a dictionary of graph names (URIRef or BNode) to Graph objects (as the graph's identifier would be removed).
+
+```
+add add_named_graph(uri: IdentifiedNode, graph: Graph) method
+add has_named_graph(uri: IdentifiedNode) method
+add remove_named_graph(uri: IdentifiedNode) method
+add replace_named_graph(uri: IdentifiedNode, graph: Graph)) method
+add graphs() method as an alias for contexts()
+add default_graph property as an alias for default_context
+add get_named_graph as an alias for get_graph
+deprecate graph(graph) method
+deprecate remove_graph(graph) method
+deprecate contexts() method
+Using IdentifiedNode as a super-interface for URIRef and BNode (since both are allowed as graph names in RDF 1.1).
+```
+
+Make the following enhancements to the triples, quads, and subject/predicate/object APIs.
+
+Major changes:
+P1. Remove `default_union` attribute and make the Dataset inclusive.
+P2. Remove the Default Graph URI ("urn:x-rdflib:default").
+P3. Remove Graph class's "identifier" attribute to align with the W3C spec, impacting Dataset methods which use the Graph class.
+P4. Make the graphs() method of Dataset return a dictionary of named graph names to Graph objects.
+Enhancements:
+P5. Support passing of iterables of Terms to triples, quads, and related methods, similar to the triples_choices method.
+P6. Default the triples method to iterate with `(None, None, None)`
+
+With all of the above changes, including those changes proposed by Martynas, here are some examples:
+
+```python
+from rdflib import Dataset, Graph, URIRef, Literal
+from rdflib.namespace import RDFS
+
+# ============================================
+# Adding Data to the Dataset
+# ============================================
+
+# Initialize the dataset
+d = Dataset()
+
+# Add a single triple to the Default Graph, and a single triple to a Named Graph
+g1 = Graph()
+g1.add(
+    (
+        URIRef("http://example.com/subject-a"),
+        URIRef("http://example.com/predicate-a"),
+        Literal("Triple A")
+    )
+)
+d.add_graph(g1)
+
+# Add a Graph to a Named Graph in the Dataset.
+g2 = Graph()
+g2.add(
+    (
+        URIRef("http://example.com/subject-b"),
+        URIRef("http://example.com/predicate-b"),
+        Literal("Triple B")
+    )
+)
+d.add_named_graph(uri=URIRef("http://example.com/graph-B"), g2)
```

Comment:
I would maybe rename `uri` to `graph_name` or `name` because RDF 1.1 allows also blank nodes to be used.

Also is it replacing the named graph or merging into it? `add` might mean both.

Reply:
"name" instead of "uri" sounds sensible, I'll update this.

The method is a merge, the proposed methods were (https://github.com/RDFLib/rdflib/discussions/2591#discussion-5619172):
```
add_named_graph(uri: IdentifiedNode, graph: Graph) method
remove_named_graph(uri: IdentifiedNode) method
replace_named_graph(uri: IdentifiedNode, graph: Graph)) method
```
We could change the method name to something like"merge_named_graph" or "add_to_named_graph". I would prefer leaving the method as "add_named_graph", so a user would need to familiarise themselves with the different methods/read the docstrings.

Reply:
updated `uri` to `name` in in 2cc3590d5c27553c774a17841c636f781fba02a4